### ISSUE TITLE
[Bug]: [Nightly]Plenty of cases failed for loading collection timeout and extend 2 more hours

### ISSUE THREAD

**Issue by NicoYuan1986** [milvus-io/milvus#24903]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 56380ea
- Deployment mode(standalone or cluster): cluster
- MQ type(rocksmq, pulsar or kafka): kafka
- SDK version(e.g. pymilvus v2.0.0rc2): 2.4.0.dev67
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Plenty of cases failed for loading collection timeout and extend 2 more hours.
seems something wrong with diskann. https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20Nightly%20CI/detail/master/403/pipeline/267
```
[2023-06-14T23:25:37.891Z] [2023-06-14 19:54:57 - ERROR - pymilvus.decorators]: RPC error: [wait_for_loading_collection], <MilvusException: (code=1, message=wait for loading collection timeout, collection: search_collection_3AVMcsQi)>, <Time:{'RPC start': '2023-06-14 19:52:57.103531', 'RPC error': '2023-06-14 19:54:57.446694'}> (decorators.py:108)
[2023-06-14T23:25:37.891Z] [2023-06-14 19:54:57 - WARNING - pymilvus.decorators]: [93mRetry timeout: 120s[0m (decorators.py:79)
[2023-06-14T23:25:37.891Z] [2023-06-14 19:54:57 - ERROR - pymilvus.decorators]: RPC error: [load_collection], <MilvusException: (code=1, message=Retry timeout: 120s, message=wait for loading collection timeout, collection: search_collection_3AVMcsQi)>, <Time:{'RPC start': '2023-06-14 19:52:56.842257', 'RPC error': '2023-06-14 19:54:57.447009'}> (decorators.py:108)
[2023-06-14T23:25:37.891Z] [2023-06-14 19:54:57 - ERROR - ci_test]: Traceback (most recent call last):
[2023-06-14T23:25:37.891Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 50, in handler
[2023-06-14T23:25:37.891Z]     return func(self, *args, **kwargs)
[2023-06-14T23:25:37.891Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 791, in load_collection
[2023-06-14T23:25:37.891Z]     self.wait_for_loading_collection(collection_name, timeout)
[2023-06-14T23:25:37.891Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 109, in handler
[2023-06-14T23:25:37.891Z]     raise e
[2023-06-14T23:25:37.891Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 105, in handler
[2023-06-14T23:25:37.891Z]     return func(*args, **kwargs)
[2023-06-14T23:25:37.891Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 136, in handler
[2023-06-14T23:25:37.891Z]     ret = func(self, *args, **kwargs)
[2023-06-14T23:25:37.891Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 85, in handler
[2023-06-14T23:25:37.891Z]     raise e
[2023-06-14T23:25:37.891Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 50, in handler
[2023-06-14T23:25:37.891Z]     return func(self, *args, **kwargs)
[2023-06-14T23:25:37.891Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 813, in wait_for_loading_collection
[2023-06-14T23:25:37.891Z]     raise MilvusException(message=f"wait for loading collection timeout, collection: {collection_name}")
[2023-06-14T23:25:37.891Z] pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=wait for loading collection timeout, collection: search_collection_3AVMcsQi)>
[2023-06-14T23:25:37.891Z] 
[2023-06-14T23:25:37.891Z] The above exception was the direct cause of the following exception:
[2023-06-14T23:25:37.891Z] 
[2023-06-14T23:25:37.891Z] Traceback (most recent call last):
[2023-06-14T23:25:37.891Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 26, in inner_wrapper
[2023-06-14T23:25:37.891Z]     res = func(*args, **_kwargs)
[2023-06-14T23:25:37.891Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 57, in api_request
[2023-06-14T23:25:37.891Z]     return func(*arg, **kwargs)
[2023-06-14T23:25:37.891Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 375, in load
[2023-06-14T23:25:37.891Z]     conn.load_collection(self._name, replica_number=replica_number, timeout=timeout, **kwargs)
[2023-06-14T23:25:37.891Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 109, in handler
[2023-06-14T23:25:37.891Z]     raise e
[2023-06-14T23:25:37.891Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 105, in handler
[2023-06-14T23:25:37.891Z]     return func(*args, **kwargs)
[2023-06-14T23:25:37.891Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 136, in handler
[2023-06-14T23:25:37.891Z]     ret = func(self, *args, **kwargs)
[2023-06-14T23:25:37.892Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 80, in handler
[2023-06-14T23:25:37.892Z]     raise MilvusException(e.code, f"{timeout_msg}, message={e.message}") from e
[2023-06-14T23:25:37.892Z] pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=Retry timeout: 120s, message=wait for loading collection timeout, collection: search_collection_3AVMcsQi)>
[2023-06-14T23:25:37.892Z]  (api_request.py:39)
[2023-06-14T23:25:37.892Z] [2023-06-14 19:54:57 - ERROR - ci_test]: (api_response) : <MilvusException: (code=1, message=Retry timeout: 120s, message=wait for loading collection timeout, collection: search_collection_3AVMcsQi)> (api_request.py:40)
```

### Expected Behavior

pass

### Milvus Log

[artifacts-milvus-distributed-kafka-nightly-403-pymilvus-e2e-logs.tar.gz](https://jenkins.milvus.io:18080/job/Milvus%20Nightly%20CI/job/master/403/artifact/artifacts-milvus-distributed-kafka-nightly-403-pymilvus-e2e-logs.tar.gz)

**Comment by xiaocai2333** [comment]
critical logs:
```
2023-06-15T03:29:58.562366195+08:00 stdout F  => Index count is 0
2023-06-15T03:29:58.562370168+08:00 stderr F E20230614 19:29:58.561400    68 diskann.cc:123] [KNOWHERE][Count][milvus] Count() function is not supported when index is not ready yet.
2023-06-15T03:29:58.562533857+08:00 stdout F [2023/06/14 19:29:58.562 +00:00] [WARN] [segments/cgo_util.go:56] ["UpdateSealedSegmentIndex failed, C Runtime Exception: [UnexpectedError] Assert \"row_count > 0\" at /go/src/github.com/milvus-io/milvus/internal/core/src/segcore/SegmentSealedImpl.cpp:81\n => Index count is 0\n"]
```

**Comment by yanliang567** [comment]
sounds like the same issue as Issue #23966? @yah01 
/assign @yah01 
/unassign

**Comment by xiaocai2333** [comment]
```
I20230615 11:14:10.037338 324423 factory.cc:20] [KNOWHERE][Create][milvus] create knowhere index DISKANN
E20230615 11:14:10.133657 324423 diskann.cc:519] [KNOWHERE][Search][milvus] Failed to load diskann.
E20230615 11:14:10.133704 324423 diskann.cc:108] [KNOWHERE][Dim][milvus] Dim() function is not supported when index is not ready yet.
E20230615 11:14:10.133801 323996 diskann.cc:123] [KNOWHERE][Count][milvus] Count() function is not supported when index is not ready yet.

```

**Comment by yah01** [comment]
Same as #23966, knowhere behavior changed
/assign @cqy123456 is fixing this

**Comment by yah01** [comment]
https://github.com/milvus-io/milvus/pull/24898

**Comment by yah01** [comment]
/assign @NicoYuan1986 
plz check with #24898

**Comment by NicoYuan1986** [comment]
Fixed! master-20230619-a6310050
